src/session/sidebar/mod.rs

```diff
@@ -201,4 +262,119 @@ impl Sidebar {
         self_.selected_chat.replace(selected_chat);
         self.notify("selected-chat");
     }
+
+    pub fn set_session(&self, session: Option<Session>) {
+        if self.session() == session {
+            return;
+        }
+
+        let self_ = imp::Sidebar::from_instance(self);
+
+        if let Some(ref session) = session {
+            // Merge ChatList and UserList into a single list model
+            let list = gio::ListStore::new(gio::ListModel::static_type());
+            list.append(session.chat_list());
+            list.append(session.user_list());
+            let model = gtk::FlattenListModel::new(Some(&list));
+
+            let filter = gtk::CustomFilter::new(
+                clone!(@strong self as obj => @default-return false, move |item| {
```

Comment:
Does this rc copy really have to be strong?

Reply:
Right! I also spotted this as well yesterday, but I forgot to fix it.